File: api/models/recommendation.py
from __future__ import annotations

from pydantic import BaseModel


class Derivative(BaseModel):
    market: str
    model_prob: float
    market_prob: float
    edge: float
    size: float
    state: str


class Feature(BaseModel):
    label: str
    value: float


class SizingStep(BaseModel):
    label: str
    value: float
    applied: bool


class VetoState(BaseModel):
    state: str
    vetoed: int
    total: int
    maps: list[str]


class MatchContext(BaseModel):
    roster_a: str
    roster_b: str
    stand_ins: str
    timezone_gap: str
    schedule: str
    news_24h: str


class StrategyHealth(BaseModel):
    clv_30d: float
    paper_roi_30d: float
    calibration_ece: float
    drift_ece: float
    days_since_refit: int


class Lineage(BaseModel):
    model: str
    model_hash: str
    feature_snapshot: str
    market_snapshot: str
    backtest_run: int


class RecommendationDetail(BaseModel):
    id: str
    match: str
    market: str
    strategy: str
    model_prob: float
    market_prob: float
    edge: float
    size: float
    confidence: str
    stake_usd: float
    close: str
    format: str
    regime: str
    veto: VetoState
    derivatives: list[Derivative]
    features: list[Feature]
    sizing_trace: list[SizingStep]
    context: MatchContext
    strategy_health: StrategyHealth
    lineage: Lineage

File: console/src/types/recommendation.ts
export interface Derivative {
  market: string
  model_prob: number
  market_prob: number
  edge: number
  size: number
  state: string
}

export interface Feature {
  label: string
  value: number
}

export interface SizingStep {
  label: string
  value: number
  applied: boolean
}

export interface VetoState {
  state: string
  vetoed: number
  total: number
  maps: string[]
}

export interface MatchContext {
  roster_a: string
  roster_b: string
  stand_ins: string
  timezone_gap: string
  schedule: string
  news_24h: string
}

export interface StrategyHealth {
  clv_30d: number
  paper_roi_30d: number
  calibration_ece: number
  drift_ece: number
  days_since_refit: number
}

export interface Lineage {
  model: string
  model_hash: string
  feature_snapshot: string
  market_snapshot: string
  backtest_run: number
}

export interface RecommendationDetail {
  id: string
  match: string
  market: string
  strategy: string
  model_prob: number
  market_prob: number
  edge: number
  size: number
  confidence: string
  stake_usd: number
  close: string
  format: string
  regime: string
  veto: VetoState
  derivatives: Derivative[]
  features: Feature[]
  sizing_trace: SizingStep[]
  context: MatchContext
  strategy_health: StrategyHealth
  lineage: Lineage
}
